src/components/input/month/index.tsx

```diff
@@ -1,46 +1,177 @@
-import { Component, Event, EventEmitter, h, Prop, Watch } from "@stencil/core"
-import { Date } from "isoly"
+import {
+	Component,
+	ComponentWillLoad,
+	Element,
+	Event,
+	EventEmitter,
+	h,
+	Host,
+	Listen,
+	Method,
+	Prop,
+	VNode,
+	Watch,
+} from "@stencil/core"
+import { isoly } from "isoly"
+import { Data } from "../../../model"
+import { Color } from "../../../model"
 import * as generate from "../../calendar/generate"
+import { Editable } from "../Editable"
+import { Input } from "../Input"
+import { Looks } from "../Looks"
 
 @Component({
 	tag: "smoothly-input-month",
 	styleUrl: "style.css",
 	scoped: true,
 })
-export class MonthSelector {
-	@Prop({ mutable: true }) value?: Date
+export class SmoothlyInputMonth implements ComponentWillLoad, Input, Editable {
+	@Element() element: HTMLSmoothlyInputMonthElement
+	@Prop({ mutable: true }) changed: boolean
+	@Prop({ reflect: true, mutable: true }) readonly: boolean
+	@Prop({ reflect: true }) color?: Color
+	@Prop({ reflect: true }) looks: Looks
+	@Prop() name: string
+	@Prop({ mutable: true }) value?: isoly.Date = isoly.Date.now()
+	@Prop({ reflect: true }) next = false
+	@Prop({ reflect: true }) previous = false
+	@Prop() arrows = true
+	@Event() smoothlyInput: EventEmitter<Data>
+	@Event() smoothlyInputLoad: EventEmitter<(parent: HTMLElement) => void>
 	@Event() smoothlyValueChange: EventEmitter<Date>
+	@Event() smoothlyFormDisable: EventEmitter<(disabled: boolean) => void>
+	@Event() smoothlyInputLooks: EventEmitter<(looks: Looks, color: Color) => void>
+	private year?: HTMLSmoothlyInputSelectElement
+	private month?: HTMLSmoothlyInputSelectElement
+	private listener: { changed?: (parent: Editable) => Promise<void> } = {}
+
+	componentWillLoad(): void {
+		this.valueChanged()
+		this.smoothlyInputLooks.emit(looks => (this.looks = looks))
+		this.smoothlyInputLoad.emit(() => {})
+		!this.readonly && this.smoothlyFormDisable.emit(readonly => (this.readonly = readonly))
+		this.valueChanged()
```

Comment:
why is this.valueChanged() run twice in componentWillLoad?

Reply:
That is an excellent question. A question i do not have the answer to.